### ISSUE TITLE
Support error filtering by level

### ISSUE THREAD

**Issue by twolfson** [twolfson/restructuredtext-lint#8]
In `docutils`, the reporters allow for filtering of errors via `report_level`. While we are circumventing this part of their system, we can reproduce this behavior.

``` python
errors = [error for error in errors if error.level <= self.report_level]
```

In `docutils`, these levels are 1 (most detailed) to 5 (no details) and can be accessed via the CLI with `--report, -r <level>`

http://repo.or.cz/w/docutils.git/blob/422cede485668203abc01c76ca317578ff634b30:/docutils/docutils/frontend.py#l429

**Comment by twolfson** [comment]
I should clarify, this is only practical for defining on the CLI. When using this on an API basis, an end user can set up their own filtering via the same list comprehension as above.

**Comment by twolfson** [comment]
This has been implemented by @peterjc in #37 and released in `0.18.0` :tada:
